Blocks/Persistence/file.h

```diff
@@ -312,6 +318,31 @@ class FilePersister {
     }
   }
 
+  std::pair<uint64_t, uint64_t> IndexRangeByTimestampRange(std::chrono::microseconds from,
+                                                           std::chrono::microseconds till) const {
+    std::pair<uint64_t, uint64_t> result{static_cast<uint64_t>(-1), static_cast<uint64_t>(-1)};
+    std::lock_guard<std::mutex> lock(file_persister_impl_->mutex);
+    const auto begin_it = std::lower_bound(
```

Comment:
Just a thought: Funny you coded it this way. I'd call `std::binary_search` and capture the `from` variable into the lambda :)